src/tools/wasm-shell.cpp

```diff
@@ -287,7 +290,11 @@ int main(int argc, const char* argv[]) {
         builders[moduleName].swap(builder);
         modules[moduleName].swap(module);
         i++;
-        assert(WasmValidator().validate(*modules[moduleName]));
+        bool valid = WasmValidator().validate(*modules[moduleName]);
+        if (!valid) {
+          WasmPrinter::printModule(modules[moduleName].get());
```

Comment:
is this correct?

Reply:
Hmm, this does change the semantics, before it would not even check if assertions are off. But I see no downside to making this change, probably better actually.